opsdir-20-2: Second half of PR addressing comments from the opsdir review (#170)
ietf-wg-asdf/SDF: sdf.md

## Definitions block

The Definitions block contains one or more groups, each identified by

> a Class Name Keyword.

Author: Right.  I took the opportunity for a slight editorial cleanup of that whole paragraph.

The keyword `sdfRequired` is provided to apply a constraint that
defines for which declarations the corresponding data are mandatory in a
grouping (sdfThing or sdfObject) modeled by the current definition.

The value of `sdfRequired` is an array of references, each indicating
one or more declarations that are mandatory to be represented.

References in this array can be SDF names (JSON Pointers), or one of
two abbreviated reference formats:

* a text string with a "referenceable-name", namely an affordance name
  or a grouping name:

>   * All affordance declarations that are directly in the same grouping

Author: Those don't work for me.  The directly means the grouping immediately enclosing them.  neither very nor exactly conveys this.  Yes, there is an i.e., but that only works with a directly in place.

>   * All affordance declarations that are directly in the same grouping

Reviewer: > Those don't work for me. The directly means the grouping immediately enclosing them. neither very nor exactly conveys this. Yes, there is an i.e., but that only works with a directly in place.

Hmm, okay, then you can just consider this comment resolved :)

Some wider security considerations applicable to Things are discussed
in {{-seccons}}.

{{Section 5 of -cddl}} gives an overview over security considerations
that arise when formal description techniques are used to govern
interoperability; analogs of these security considerations can apply
to SDF.

The security considerations of underlying building blocks such as
those detailed in {{Section 10 of RFC3629@-utf8}} apply.

SDF uses JSON as a representation language.
For a number of
cases {{-json}} indicates that implementation behavior for certain constructs
allowed by the JSON grammar is unpredictable.

Implementations need to be robust against invalid or unpredictable
cases on input, preferably by rejecting input that is invalid or
that would lead to unpredictable behavior, and need to avoid generating
these cases on output.

Implementations of model languages may also exhibit
performance-related availability issues when the attacker can control
the input, see {{Section 4.1 of -jsonpath}} for a brief discussion and

> {{Section 8 of -iregexp}} for considerations specifically about the use
> of `pattern`.

Reviewer: I stumbled a bit over this formulation, but I am not sure whether this is better/more correct.

```suggestion
{{Section 8 of -iregexp}} for considerations specific to the use
of `pattern`.
```
